app/gateway/2.8.x/plan-and-deploy/security/secrets-management/backends/hashicorp-vault.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Hashicorp Vault
+badge: enterprise
+---
+
+## Configuration
+
+[Hashicorp Vault](https://www.vaultproject.io/) can be configured with environment variables or with a Vault entity.
+## Environment variables
+
+```bash
+export KONG_VAULTS_HCV_PROTOCOL=<protocol(http|https)>
+export KONG_VAULTS_HCV_HOST=<hostname>
+export KONG_VAULTS_HCV_PORT=<portnumber>
+export KONG_VAULTS_HCV_MOUNT=<mountpoint>
+export KONG_VAULTS_HCV_KV=<v1|v2>
+export KONG_VAULTS_HCV_TOKEN=<tokenstring>
+```
+
+You can also store this information in an entity.
+
+## Entity
+
+The Vault entity can only be used once the database is initialized. Secrets for values that are used _before_ the database is initialized can't make use of the Vaults entity.
+
+{% navtabs codeblock %}
+{% navtab cURL %}
+
+```bash
+curl -i -X PUT http://<hostname>:8001/vaults-beta/my-hashicorp-vault \
+  --data name="hcv" \
+  --data description="Storing secrets in Hashicorp Vault" \
+  --data config.protocol="https" \
+  --data config.host="localhost" \
+  --data config.port="8200" \
+  --data config.mount="vault" \
+  --data config.kv="v2" \
+  --data config.token="<mytoken>"
+```
```

Comment:
It feels like these configs are not totally in-line with examples below. E.g. that `mount` should probably be `secret` (perhaps that needs to be the default in schema?), which also means that you don't need to repeat the mount in reference as it is prefixed by default:

https://github.com/Kong/kong-ee/blob/master/kong/vaults/hcv/init.lua#L54